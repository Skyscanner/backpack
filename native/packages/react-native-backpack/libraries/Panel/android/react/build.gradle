apply plugin: 'com.android.library'
apply plugin: 'kotlin-android'

android {
  compileSdkVersion rootProject.ext.compileSdkVersion
  buildToolsVersion rootProject.ext.buildToolsVersion


  defaultConfig {
    minSdkVersion rootProject.ext.minSdkVersion
    targetSdkVersion rootProject.ext.targetSdkVersion
    versionCode 1
    versionName "1.0"
  }
}

dependencies{
  implementation "com.facebook.react:react-native:+" // From node_modules
  implementation project(':panel-native')
  implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
}
repositories {
  mavenCentral()
}
