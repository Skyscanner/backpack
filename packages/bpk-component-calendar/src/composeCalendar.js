/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2017 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import PropTypes from 'prop-types';
import React from 'react';
import { cssModules } from 'bpk-react-utils';

import CustomPropTypes from './custom-proptypes';
import STYLES from './bpk-calendar.scss';

const getClassName = cssModules(STYLES);

const composeCalendar = (Nav, GridHeader, Grid, CalendarDate) => {
  const BpkCalendar = (props) => {
    const classNames = [getClassName('bpk-calendar')];

    const {
      changeMonthLabel,
      className,
      dateModifiers,
      daysOfWeek,
      focusedDate,
      formatDateFull,
      formatMonth,
      id,
      markOutsideDays,
      markToday,
      maxDate,
      minDate,
      month,
      onDateClick,
      onDateMouseDown,
      onDateKeyDown,
      onMonthChange,
      preventKeyboardFocus,
      selectedDate,
      showWeekendSeparator,
      weekStartsOn,
      fixedWidth,
    } = props;

    if (className) { classNames.push(className); }
    if (fixedWidth) { classNames.push(getClassName('bpk-calendar--fixed')); }

    const headerClasses = [];
    // If the Nav is present add `bpk-calendar__header` which
    // adds spacing between the nav and header.
    if (Nav) { headerClasses.push(getClassName('bpk-calendar__header')); }

    const gridClasses = [];
    // If the GridHeader is not present add `bpk-calendar__grid` which
    // adds spacing between the nav and grid.
    if (!GridHeader && Nav) { gridClasses.push(getClassName('bpk-calendar__grid')); }

    return (
      <div className={classNames.join(' ')}>
        { Nav && (
          <Nav
            changeMonthLabel={changeMonthLabel}
            formatMonth={formatMonth}
            id={`${id}__bpk_calendar_nav`}
            maxDate={maxDate}
            minDate={minDate}
            month={month}
            onMonthChange={onMonthChange}
          />
        )}
        { GridHeader && (
          <GridHeader
            daysOfWeek={daysOfWeek}
            showWeekendSeparator={showWeekendSeparator}
            weekStartsOn={weekStartsOn}
            className={headerClasses.join(' ')}
          />
        )}
        <Grid
          DateComponent={CalendarDate}
          dateModifiers={dateModifiers}
          daysOfWeek={daysOfWeek}
          formatDateFull={formatDateFull}
          formatMonth={formatMonth}
          month={month}
          onDateClick={onDateClick}
          onDateMouseDown={onDateMouseDown}
          onDateKeyDown={onDateKeyDown}
          preventKeyboardFocus={preventKeyboardFocus}
          showWeekendSeparator={showWeekendSeparator}
          weekStartsOn={weekStartsOn}
          maxDate={maxDate}
          minDate={minDate}
          focusedDate={focusedDate}
          markToday={markToday}
          markOutsideDays={markOutsideDays}
          selectedDate={selectedDate}
          className={gridClasses.join(' ')}
        />
      </div>
    );
  };

  BpkCalendar.propTypes = {
    // Required
    changeMonthLabel: PropTypes.string.isRequired,
    daysOfWeek: CustomPropTypes.DaysOfWeek.isRequired,
    formatDateFull: PropTypes.func.isRequired,
    formatMonth: PropTypes.func.isRequired,
    id: PropTypes.string.isRequired,
    maxDate: PropTypes.instanceOf(Date).isRequired,
    minDate: PropTypes.instanceOf(Date).isRequired,
    month: PropTypes.instanceOf(Date).isRequired,
    // Optional
    className: PropTypes.string,
    dateModifiers: CustomPropTypes.DateModifiers,
    fixedWidth: PropTypes.bool,
    focusedDate: PropTypes.instanceOf(Date),
    markOutsideDays: PropTypes.bool,
    markToday: PropTypes.bool,
    onMonthChange: PropTypes.func,
    onDateClick: PropTypes.func,
    onDateMouseDown: PropTypes.func,
    onDateKeyDown: PropTypes.func,
    preventKeyboardFocus: PropTypes.bool,
    selectedDate: PropTypes.instanceOf(Date),
    showWeekendSeparator: PropTypes.bool,
    weekStartsOn: PropTypes.number,
  };

  BpkCalendar.defaultProps = {
    className: null,
    dateModifiers: {},
    fixedWidth: true,
    focusedDate: null,
    markOutsideDays: true,
    markToday: true,
    onMonthChange: () => null,
    onDateClick: () => null,
    onDateMouseDown: () => null,
    onDateKeyDown: () => null,
    preventKeyboardFocus: false,
    selectedDate: null,
    showWeekendSeparator: true,
    weekStartsOn: 1,
  };

  return BpkCalendar;
};

export default composeCalendar;
