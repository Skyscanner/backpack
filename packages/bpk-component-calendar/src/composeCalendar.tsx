/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import type { ComponentType } from 'react';

import { cssModules } from '../../bpk-react-utils';

import { CALENDAR_SELECTION_TYPE } from './custom-proptypes';
import { memoize } from './utils';

import type {
  DaysOfWeek,
  SelectionConfiguration,
  DateModifiers,
} from './custom-proptypes';

import STYLES from './BpkCalendar.module.scss';

const getClassName = cssModules(STYLES);

export type Props = {
  /**
   * **Required** if Nav !== null
   */
  changeMonthLabel?: string | null;
  /**
   * An array of objects describing the days of the week
   */
  daysOfWeek: DaysOfWeek;
  /**
   * A function to format a full, human-readable date, for example: "Monday, 8th January 2018".
   */
  formatDateFull: (date: Date) => Date | string;
  /**
   * A function to format a human-readable month, for example: "January 2018":
   * If you just need to quickly prototype, use the following from [`date-fns`](https://date-fns.org/docs/format#usage)
   */
  formatMonth: (date: Date) => Date | string;
  id: string;
  maxDate: Date;
  minDate: Date;
  month: Date;
  /**
   * **Required** if Nav !== null
   */
  nextMonthLabel?: string | null;
  /**
   * **Required** if Nav !== null
   */
  previousMonthLabel?: string | null;
  /**
   * First day of the week. 0 = Sunday, 1 = Monday, ..., 6 = Saturday.
   */
  weekStartsOn: 0 | 1 | 2 | 3 | 4 | 5 | 6;
  className?: string | null;
  dateModifiers?: DateModifiers;
  /**
   * If set to true (default), it sets a fixed width on the calendar container. This is necessary to support transitions and to create the right size for the Datepicker component.
   * If set to false, the calendar is of fluid width and will take up the space of its parent container.
   */
  fixedWidth?: boolean;
  focusedDate?: Date | null;
  markOutsideDays?: boolean;
  markToday?: boolean;
  onMonthChange?:
    | ((
        event: UIEvent,
        { month, source }: { month: Date; source: string },
      ) => void)
    | null;
  onDateClick?: ((date: Date) => void) | null;
  onDateKeyDown?: ((event: KeyboardEvent) => void) | null;
  preventKeyboardFocus?: boolean;
  selectionConfiguration?: SelectionConfiguration;
  gridClassName?: string | null;
  /**
   * Sets the height of month rows in 'rem' units. If not specified, the default value of `2.75rem` will be used.
   */
  customRowHeight?: number;
  /**
   * Key to be used to pick the desired weekDay format from the `daysOfWeek` object, for example: `nameAbbr` or `nameNarrow`.
   */
  weekDayKey?: string;
  /**
   * This is useful if your custom implementation of Nav component requires additional properties. They will be passed, unmodified.
   */
  navProps?: {} | null;
  /**
   * This is useful if your custom implementation of Header component requires additional properties. They will be passed, unmodified.
   */
  headerProps?: {} | null;
  /**
   * This is useful if your custom implementation of Grid component requires additional properties. They will be passed, unmodified.
   */
  gridProps?: {} | null;
  /**
   * This is useful if your custom implementation of Date component requires additional properties. They will be passed, unmodified.
   */
  dateProps?: {} | null;
};

const composeCalendar = (
  Nav: ComponentType<any> | string | null,
  GridHeader: ComponentType<any> | string | null,
  Grid: ComponentType<any> | string,
  CalendarDate: ComponentType<any> | string | null,
) => {
  const BpkCalendar = ({
    changeMonthLabel = null,
    className = null,
    customRowHeight,
    dateModifiers = {},
    dateProps = {},
    daysOfWeek,
    fixedWidth = true,
    focusedDate = null,
    formatDateFull,
    formatMonth,
    gridClassName = null,
    gridProps = {},
    headerProps = {},
    id,
    markOutsideDays = true,
    markToday = true,
    maxDate,
    minDate,
    month,
    navProps = {},
    nextMonthLabel = null,
    onDateClick = null,
    onDateKeyDown = null,
    onMonthChange = null,
    preventKeyboardFocus = false,
    previousMonthLabel = null,
    selectionConfiguration = {
      type: CALENDAR_SELECTION_TYPE.single,
      date: null,
    },
    weekDayKey = 'nameAbbr',
    weekStartsOn,
  }: Props) => {
    const classNames = [getClassName('bpk-calendar')];

    if (className) {
      classNames.push(className);
    }
    if (fixedWidth) {
      classNames.push(getClassName('bpk-calendar--fixed'));
    }

    const headerClasses = [];
    // If the Nav is present add `bpk-calendar__header` which
    // adds spacing between the nav and header.
    if (Nav) {
      headerClasses.push(getClassName('bpk-calendar__header'));
    }

    const gridClasses = [];
    // If the GridHeader is not present add `bpk-calendar__grid` which
    // adds spacing between the nav and grid.
    if (!GridHeader && Nav) {
      gridClasses.push(getClassName('bpk-calendar__grid'));
    }
    if (gridClassName) {
      gridClasses.push(gridClassName);
    }

    return (
      <div className={classNames.join(' ')}>
        {Nav && (
          <Nav
            changeMonthLabel={changeMonthLabel}
            formatMonth={memoize(formatMonth)}
            id={`${id}__bpk_calendar_nav`}
            maxDate={maxDate}
            minDate={minDate}
            month={month}
            nextMonthLabel={nextMonthLabel}
            onMonthChange={onMonthChange}
            previousMonthLabel={previousMonthLabel}
            {...navProps}
          />
        )}
        {GridHeader && (
          <GridHeader
            daysOfWeek={daysOfWeek}
            weekStartsOn={weekStartsOn}
            weekDayKey={weekDayKey}
            // TODO: className to be removed
            // eslint-disable-next-line @skyscanner/rules/forbid-component-props
            className={headerClasses.join(' ')}
            {...headerProps}
          />
        )}
        <Grid
          DateComponent={CalendarDate}
          dateModifiers={dateModifiers}
          daysOfWeek={daysOfWeek}
          formatDateFull={memoize(formatDateFull)}
          formatMonth={memoize(formatMonth)}
          month={month}
          onDateClick={onDateClick}
          onDateKeyDown={onDateKeyDown}
          preventKeyboardFocus={preventKeyboardFocus}
          weekStartsOn={weekStartsOn}
          maxDate={maxDate}
          minDate={minDate}
          focusedDate={focusedDate}
          markToday={markToday}
          markOutsideDays={markOutsideDays}
          // TODO: className to be removed
          // eslint-disable-next-line @skyscanner/rules/forbid-component-props
          className={gridClasses.join(' ')}
          dateProps={dateProps}
          selectionConfiguration={selectionConfiguration}
          customRowHeight={customRowHeight}
          {...gridProps}
        />
      </div>
    );
  };

  return BpkCalendar;
};

export default composeCalendar;
