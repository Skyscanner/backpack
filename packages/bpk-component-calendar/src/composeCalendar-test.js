/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { render } from '@testing-library/react';

import { weekDays, formatMonth, formatDateFull } from '../test-utils';

import composeCalendar from './composeCalendar';

const createNodeMock = () => ({
  focus: () => null,
});

const CalendarComponent = composeCalendar('Nav', 'Header', 'Grid', 'Date');

const CustomCalendarComponent = composeCalendar(null, null, 'Grid', 'Date');

const HeaderAndGridCalendar = composeCalendar('Nav', null, 'Grid', 'Date');

describe('composeCalendar', () => {
  it('should compose a nav, header, grid and date component correctly', () => {
    const { asFragment } = render(
      <CalendarComponent
        id="myCalendar"
        formatMonth={formatMonth}
        formatDateFull={formatDateFull}
        daysOfWeek={weekDays}
        weekStartsOn={1}
        changeMonthLabel="Change month"
        previousMonthLabel="Go to previous month"
        nextMonthLabel="Go to next month"
        minDate={new Date(Date.UTC(2010, 1, 15))}
        maxDate={new Date(Date.UTC(2010, 2, 15))}
        month={new Date(Date.UTC(2010, 1, 15))}
      />,
      { createNodeMock },
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with custom className prop correctly', () => {
    const { asFragment } = render(
      <CalendarComponent
        id="myCalendar"
        formatMonth={formatMonth}
        formatDateFull={formatDateFull}
        daysOfWeek={weekDays}
        weekStartsOn={1}
        changeMonthLabel="Change month"
        previousMonthLabel="Go to previous month"
        nextMonthLabel="Go to next month"
        minDate={new Date(Date.UTC(2010, 1, 15))}
        maxDate={new Date(Date.UTC(2010, 2, 15))}
        month={new Date(Date.UTC(2010, 1, 15))}
        className="my-custom-class"
      />,
      { createNodeMock },
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with "fixedWidth" set to false', () => {
    const { asFragment } = render(
      <CalendarComponent
        id="myCalendar"
        formatMonth={formatMonth}
        formatDateFull={formatDateFull}
        daysOfWeek={weekDays}
        weekStartsOn={1}
        changeMonthLabel="Change month"
        previousMonthLabel="Go to previous month"
        nextMonthLabel="Go to next month"
        minDate={new Date(Date.UTC(2010, 1, 15))}
        maxDate={new Date(Date.UTC(2010, 2, 15))}
        month={new Date(Date.UTC(2010, 1, 15))}
        fixedWidth={false}
      />,
      { createNodeMock },
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render without a nav or header element', () => {
    const { asFragment } = render(
      <CustomCalendarComponent
        id="myCalendar"
        formatMonth={formatMonth}
        formatDateFull={formatDateFull}
        daysOfWeek={weekDays}
        weekStartsOn={1}
        changeMonthLabel="Change month"
        previousMonthLabel="Go to previous month"
        nextMonthLabel="Go to next month"
        minDate={new Date(Date.UTC(2010, 1, 15))}
        maxDate={new Date(Date.UTC(2010, 2, 15))}
        month={new Date(Date.UTC(2010, 1, 15))}
      />,
      { createNodeMock },
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render without a header element', () => {
    const { asFragment } = render(
      <HeaderAndGridCalendar
        id="myCalendar"
        formatMonth={formatMonth}
        formatDateFull={formatDateFull}
        daysOfWeek={weekDays}
        weekStartsOn={1}
        changeMonthLabel="Change month"
        previousMonthLabel="Go to previous month"
        nextMonthLabel="Go to next month"
        minDate={new Date(Date.UTC(2010, 1, 15))}
        maxDate={new Date(Date.UTC(2010, 2, 15))}
        month={new Date(Date.UTC(2010, 1, 15))}
      />,
      { createNodeMock },
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should pass props to their respective components', () => {
    const PropsBasedCalendarComponent = composeCalendar(
      (props) => props.name,
      (props) => props.name,
      (props) => (
        <div>
          <h1>{props.name}</h1>
          <props.DateComponent {...props.dateProps} />
        </div>
      ),
      (props) => props.name,
    );

    const { asFragment } = render(
      <PropsBasedCalendarComponent
        id="myCalendar"
        formatMonth={formatMonth}
        formatDateFull={formatDateFull}
        daysOfWeek={weekDays}
        weekStartsOn={1}
        changeMonthLabel="Change month"
        previousMonthLabel="Go to previous month"
        nextMonthLabel="Go to next month"
        minDate={new Date(Date.UTC(2010, 1, 15))}
        maxDate={new Date(Date.UTC(2010, 2, 15))}
        month={new Date(Date.UTC(2010, 1, 15))}
        navProps={{ name: 'Nav' }}
        headerProps={{ name: 'Header' }}
        gridProps={{ name: 'Grid' }}
        dateProps={{ name: 'Date' }}
      />,
      { createNodeMock },
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
