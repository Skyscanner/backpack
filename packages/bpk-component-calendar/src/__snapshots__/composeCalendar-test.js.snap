// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`composeCalendar should compose a nav, header, grid and date component correctly 1`] = `
<div
  className="bpk-calendar bpk-calendar--fixed"
>
  <Nav
    changeMonthLabel="Change month"
    formatMonth={[Function]}
    id="myCalendar__bpk_calendar_nav"
    maxDate={2010-03-15T00:00:00.000Z}
    minDate={2010-02-15T00:00:00.000Z}
    month={2010-02-15T00:00:00.000Z}
    onMonthChange={[Function]}
  />
  <Header
    className="bpk-calendar__header"
    daysOfWeek={
      Array [
        Object {
          "index": 0,
          "isWeekend": true,
          "name": "Sunday",
          "nameAbbr": "Sun",
        },
        Object {
          "index": 1,
          "isWeekend": false,
          "name": "Monday",
          "nameAbbr": "Mon",
        },
        Object {
          "index": 2,
          "isWeekend": false,
          "name": "Tuesday",
          "nameAbbr": "Tue",
        },
        Object {
          "index": 3,
          "isWeekend": false,
          "name": "Wednesday",
          "nameAbbr": "Wed",
        },
        Object {
          "index": 4,
          "isWeekend": false,
          "name": "Thursday",
          "nameAbbr": "Thu",
        },
        Object {
          "index": 5,
          "isWeekend": false,
          "name": "Friday",
          "nameAbbr": "Fri",
        },
        Object {
          "index": 6,
          "isWeekend": true,
          "name": "Saturday",
          "nameAbbr": "Sat",
        },
      ]
    }
    showWeekendSeparator={true}
    weekStartsOn={1}
  />
  <Grid
    DateComponent="Date"
    className=""
    dateModifiers={Object {}}
    daysOfWeek={
      Array [
        Object {
          "index": 0,
          "isWeekend": true,
          "name": "Sunday",
          "nameAbbr": "Sun",
        },
        Object {
          "index": 1,
          "isWeekend": false,
          "name": "Monday",
          "nameAbbr": "Mon",
        },
        Object {
          "index": 2,
          "isWeekend": false,
          "name": "Tuesday",
          "nameAbbr": "Tue",
        },
        Object {
          "index": 3,
          "isWeekend": false,
          "name": "Wednesday",
          "nameAbbr": "Wed",
        },
        Object {
          "index": 4,
          "isWeekend": false,
          "name": "Thursday",
          "nameAbbr": "Thu",
        },
        Object {
          "index": 5,
          "isWeekend": false,
          "name": "Friday",
          "nameAbbr": "Fri",
        },
        Object {
          "index": 6,
          "isWeekend": true,
          "name": "Saturday",
          "nameAbbr": "Sat",
        },
      ]
    }
    focusedDate={null}
    formatDateFull={[Function]}
    formatMonth={[Function]}
    markOutsideDays={true}
    markToday={true}
    maxDate={2010-03-15T00:00:00.000Z}
    minDate={2010-02-15T00:00:00.000Z}
    month={2010-02-15T00:00:00.000Z}
    onDateClick={[Function]}
    onDateKeyDown={[Function]}
    onDateMouseDown={[Function]}
    preventKeyboardFocus={false}
    selectedDate={null}
    showWeekendSeparator={true}
    weekStartsOn={1}
  />
</div>
`;

exports[`composeCalendar should render with "fixedWidth" set to false 1`] = `
<div
  className="bpk-calendar"
>
  <Nav
    changeMonthLabel="Change month"
    formatMonth={[Function]}
    id="myCalendar__bpk_calendar_nav"
    maxDate={2010-03-15T00:00:00.000Z}
    minDate={2010-02-15T00:00:00.000Z}
    month={2010-02-15T00:00:00.000Z}
    onMonthChange={[Function]}
  />
  <Header
    className="bpk-calendar__header"
    daysOfWeek={
      Array [
        Object {
          "index": 0,
          "isWeekend": true,
          "name": "Sunday",
          "nameAbbr": "Sun",
        },
        Object {
          "index": 1,
          "isWeekend": false,
          "name": "Monday",
          "nameAbbr": "Mon",
        },
        Object {
          "index": 2,
          "isWeekend": false,
          "name": "Tuesday",
          "nameAbbr": "Tue",
        },
        Object {
          "index": 3,
          "isWeekend": false,
          "name": "Wednesday",
          "nameAbbr": "Wed",
        },
        Object {
          "index": 4,
          "isWeekend": false,
          "name": "Thursday",
          "nameAbbr": "Thu",
        },
        Object {
          "index": 5,
          "isWeekend": false,
          "name": "Friday",
          "nameAbbr": "Fri",
        },
        Object {
          "index": 6,
          "isWeekend": true,
          "name": "Saturday",
          "nameAbbr": "Sat",
        },
      ]
    }
    showWeekendSeparator={true}
    weekStartsOn={1}
  />
  <Grid
    DateComponent="Date"
    className=""
    dateModifiers={Object {}}
    daysOfWeek={
      Array [
        Object {
          "index": 0,
          "isWeekend": true,
          "name": "Sunday",
          "nameAbbr": "Sun",
        },
        Object {
          "index": 1,
          "isWeekend": false,
          "name": "Monday",
          "nameAbbr": "Mon",
        },
        Object {
          "index": 2,
          "isWeekend": false,
          "name": "Tuesday",
          "nameAbbr": "Tue",
        },
        Object {
          "index": 3,
          "isWeekend": false,
          "name": "Wednesday",
          "nameAbbr": "Wed",
        },
        Object {
          "index": 4,
          "isWeekend": false,
          "name": "Thursday",
          "nameAbbr": "Thu",
        },
        Object {
          "index": 5,
          "isWeekend": false,
          "name": "Friday",
          "nameAbbr": "Fri",
        },
        Object {
          "index": 6,
          "isWeekend": true,
          "name": "Saturday",
          "nameAbbr": "Sat",
        },
      ]
    }
    focusedDate={null}
    formatDateFull={[Function]}
    formatMonth={[Function]}
    markOutsideDays={true}
    markToday={true}
    maxDate={2010-03-15T00:00:00.000Z}
    minDate={2010-02-15T00:00:00.000Z}
    month={2010-02-15T00:00:00.000Z}
    onDateClick={[Function]}
    onDateKeyDown={[Function]}
    onDateMouseDown={[Function]}
    preventKeyboardFocus={false}
    selectedDate={null}
    showWeekendSeparator={true}
    weekStartsOn={1}
  />
</div>
`;

exports[`composeCalendar should render with custom className prop correctly 1`] = `
<div
  className="bpk-calendar my-custom-class bpk-calendar--fixed"
>
  <Nav
    changeMonthLabel="Change month"
    formatMonth={[Function]}
    id="myCalendar__bpk_calendar_nav"
    maxDate={2010-03-15T00:00:00.000Z}
    minDate={2010-02-15T00:00:00.000Z}
    month={2010-02-15T00:00:00.000Z}
    onMonthChange={[Function]}
  />
  <Header
    className="bpk-calendar__header"
    daysOfWeek={
      Array [
        Object {
          "index": 0,
          "isWeekend": true,
          "name": "Sunday",
          "nameAbbr": "Sun",
        },
        Object {
          "index": 1,
          "isWeekend": false,
          "name": "Monday",
          "nameAbbr": "Mon",
        },
        Object {
          "index": 2,
          "isWeekend": false,
          "name": "Tuesday",
          "nameAbbr": "Tue",
        },
        Object {
          "index": 3,
          "isWeekend": false,
          "name": "Wednesday",
          "nameAbbr": "Wed",
        },
        Object {
          "index": 4,
          "isWeekend": false,
          "name": "Thursday",
          "nameAbbr": "Thu",
        },
        Object {
          "index": 5,
          "isWeekend": false,
          "name": "Friday",
          "nameAbbr": "Fri",
        },
        Object {
          "index": 6,
          "isWeekend": true,
          "name": "Saturday",
          "nameAbbr": "Sat",
        },
      ]
    }
    showWeekendSeparator={true}
    weekStartsOn={1}
  />
  <Grid
    DateComponent="Date"
    className=""
    dateModifiers={Object {}}
    daysOfWeek={
      Array [
        Object {
          "index": 0,
          "isWeekend": true,
          "name": "Sunday",
          "nameAbbr": "Sun",
        },
        Object {
          "index": 1,
          "isWeekend": false,
          "name": "Monday",
          "nameAbbr": "Mon",
        },
        Object {
          "index": 2,
          "isWeekend": false,
          "name": "Tuesday",
          "nameAbbr": "Tue",
        },
        Object {
          "index": 3,
          "isWeekend": false,
          "name": "Wednesday",
          "nameAbbr": "Wed",
        },
        Object {
          "index": 4,
          "isWeekend": false,
          "name": "Thursday",
          "nameAbbr": "Thu",
        },
        Object {
          "index": 5,
          "isWeekend": false,
          "name": "Friday",
          "nameAbbr": "Fri",
        },
        Object {
          "index": 6,
          "isWeekend": true,
          "name": "Saturday",
          "nameAbbr": "Sat",
        },
      ]
    }
    focusedDate={null}
    formatDateFull={[Function]}
    formatMonth={[Function]}
    markOutsideDays={true}
    markToday={true}
    maxDate={2010-03-15T00:00:00.000Z}
    minDate={2010-02-15T00:00:00.000Z}
    month={2010-02-15T00:00:00.000Z}
    onDateClick={[Function]}
    onDateKeyDown={[Function]}
    onDateMouseDown={[Function]}
    preventKeyboardFocus={false}
    selectedDate={null}
    showWeekendSeparator={true}
    weekStartsOn={1}
  />
</div>
`;

exports[`composeCalendar should render without a header element 1`] = `
<div
  className="bpk-calendar bpk-calendar--fixed"
>
  <Nav
    changeMonthLabel="Change month"
    formatMonth={[Function]}
    id="myCalendar__bpk_calendar_nav"
    maxDate={2010-03-15T00:00:00.000Z}
    minDate={2010-02-15T00:00:00.000Z}
    month={2010-02-15T00:00:00.000Z}
    onMonthChange={[Function]}
  />
  <Grid
    DateComponent="Date"
    className="bpk-calendar__grid"
    dateModifiers={Object {}}
    daysOfWeek={
      Array [
        Object {
          "index": 0,
          "isWeekend": true,
          "name": "Sunday",
          "nameAbbr": "Sun",
        },
        Object {
          "index": 1,
          "isWeekend": false,
          "name": "Monday",
          "nameAbbr": "Mon",
        },
        Object {
          "index": 2,
          "isWeekend": false,
          "name": "Tuesday",
          "nameAbbr": "Tue",
        },
        Object {
          "index": 3,
          "isWeekend": false,
          "name": "Wednesday",
          "nameAbbr": "Wed",
        },
        Object {
          "index": 4,
          "isWeekend": false,
          "name": "Thursday",
          "nameAbbr": "Thu",
        },
        Object {
          "index": 5,
          "isWeekend": false,
          "name": "Friday",
          "nameAbbr": "Fri",
        },
        Object {
          "index": 6,
          "isWeekend": true,
          "name": "Saturday",
          "nameAbbr": "Sat",
        },
      ]
    }
    focusedDate={null}
    formatDateFull={[Function]}
    formatMonth={[Function]}
    markOutsideDays={true}
    markToday={true}
    maxDate={2010-03-15T00:00:00.000Z}
    minDate={2010-02-15T00:00:00.000Z}
    month={2010-02-15T00:00:00.000Z}
    onDateClick={[Function]}
    onDateKeyDown={[Function]}
    onDateMouseDown={[Function]}
    preventKeyboardFocus={false}
    selectedDate={null}
    showWeekendSeparator={true}
    weekStartsOn={1}
  />
</div>
`;

exports[`composeCalendar should render without a nav or header element 1`] = `
<div
  className="bpk-calendar bpk-calendar--fixed"
>
  <Grid
    DateComponent="Date"
    className=""
    dateModifiers={Object {}}
    daysOfWeek={
      Array [
        Object {
          "index": 0,
          "isWeekend": true,
          "name": "Sunday",
          "nameAbbr": "Sun",
        },
        Object {
          "index": 1,
          "isWeekend": false,
          "name": "Monday",
          "nameAbbr": "Mon",
        },
        Object {
          "index": 2,
          "isWeekend": false,
          "name": "Tuesday",
          "nameAbbr": "Tue",
        },
        Object {
          "index": 3,
          "isWeekend": false,
          "name": "Wednesday",
          "nameAbbr": "Wed",
        },
        Object {
          "index": 4,
          "isWeekend": false,
          "name": "Thursday",
          "nameAbbr": "Thu",
        },
        Object {
          "index": 5,
          "isWeekend": false,
          "name": "Friday",
          "nameAbbr": "Fri",
        },
        Object {
          "index": 6,
          "isWeekend": true,
          "name": "Saturday",
          "nameAbbr": "Sat",
        },
      ]
    }
    focusedDate={null}
    formatDateFull={[Function]}
    formatMonth={[Function]}
    markOutsideDays={true}
    markToday={true}
    maxDate={2010-03-15T00:00:00.000Z}
    minDate={2010-02-15T00:00:00.000Z}
    month={2010-02-15T00:00:00.000Z}
    onDateClick={[Function]}
    onDateKeyDown={[Function]}
    onDateMouseDown={[Function]}
    preventKeyboardFocus={false}
    selectedDate={null}
    showWeekendSeparator={true}
    weekStartsOn={1}
  />
</div>
`;
